### PR TITLE
Consolidated cases docket number refactor

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/business/useCaseHelper/consolidatedCases/processUserAssociatedCases.test.js [hunk 1]
```diff
@@ -1,98 +1,73 @@
...
-    applicationContext
-      .getPersistenceGateway()
-      .getCaseByDocketNumber.mockReturnValue(MOCK_CASE);
-  });
-
   beforeEach(() => {
     mockFoundCasesList = [MOCK_CASE];
   });
 
   it('should set isRequestingUserAssociated to true for each case', async () => {
```
Comment (rachelschneiderman) [inline shared/src/business/useCaseHelper/consolidatedCases/processUserAssociatedCases.test.js:11]: Nit: I think these tests no longer need 'async'?

Comment (rachaelparris) [reply shared/src/business/useCaseHelper/consolidatedCases/processUserAssociatedCases.test.js:11]: Good catch! I will add that to my next PR.

File: shared/src/business/useCases/caseConsolidation/addConsolidatedCaseInteractor.test.js [hunk 2]
```diff
@@ -94,30 +81,30 @@ describe('addConsolidatedCaseInteractor', () => {
...
     ).rejects.toThrow('Unauthorized for case consolidation');
   });
 
-  it('Should try to get the case by its caseId', async () => {
+  it('Should try to get the case by its docketNumber', async () => {
     await addConsolidatedCaseInteractor({
       applicationContext,
-      caseIdToConsolidateWith: 'd44ba5a9-b37b-479d-9201-067ec6e335aa', // docketNumber: '319-19'
       docketNumber: '519-19',
+      docketNumberToConsolidateWith: '319-19',
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/caseConsolidation/addConsolidatedCaseInteractor.test.js:94]: These tests are so much easier to read now!

**Review by kkoskelin — APPROVED** [review]

File: web-api/migrations/00016-lead-docket-number.test.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    expect(putStub.mock.calls[0][0]['Item'].leadDocketNumber).toEqual(
+      LEAD_DOCKET_NUMBER,
+    );
+    expect(putStub.mock.calls[1][0]['Item'].leadDocketNumber).toEqual(
+      LEAD_DOCKET_NUMBER,
+    );
+  });
+
+  it('should not add leadDocketNumber if item is a case or user-case record and has a leadCaseId but retrieving the lead case does not return a docketNumber', async () => {
```
Comment (rachelschneiderman) [inline web-api/migrations/00016-lead-docket-number.test.js:112]: Just for my understanding, why would the caseRecord that's retrieved not have a docketNumber?

Comment (rachaelparris) [reply web-api/migrations/00016-lead-docket-number.test.js:112]: Only if leadCaseId is a caseId for a case that does not exist anymore, or it was set incorrectly in the first place. This is probably overly cautious on my part.

**Review by rachelschneiderman — APPROVED** [review]

File: web-api/migrations/00016-lead-docket-number.test.js (hunk 3 above)
Comment (rachelschneiderman) [reply web-api/migrations/00016-lead-docket-number.test.js:112]: Ok, gotcha! Thanks!

**Pull request merged**